pkg/kubelet/dockertools/docker.go

```diff
@@ -1049,3 +1049,21 @@ func RunContainer(client DockerInterface, container *api.Container, pod *api.Pod
 	}
 	return dockerContainer.ID, nil
 }
+
+// KillContainer kills a docker container with the given containerID.
+// TODO(yifan): To use strong type for the containerID.
+func KillContainer(client DockerInterface, containerID string,
```

Comment:
@vmarmol I got your point. The reason here is if we want to let container runtime to hide the pod infra container behind the interface, we will need to kill and restart the whole pod in the runtime. So we need to be able to manage the readiness, and report events.

There could be other approaches that can leave the kubelet to manage the readiness, ref, etc. 
One came to my mind is to add something like `preparePod` for the kubelet. After entering into `syncPod`, `preparePod` will be invoked first:
```
switch container runtime choice:
    case docker: check and compare the pod infra container, kill everything if necessary
    case rocket: no-op
    case other: ...
```